service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -94,10 +95,31 @@ public void initializeSession(IoSession session, ConnectFuture future) {
             HttpConnectSession connectSession = (HttpConnectSession) session;
             connectSession.setVersion(acceptSession.getVersion());
             connectSession.setMethod(acceptSession.getMethod());
-            connectSession.setRequestURI(acceptSession.getRequestURI());
+            if (!validateRequestPath()) {
```

Comment:
Can you see what happens to the future object in this method ? I need to check whether we need to fail the ConnectFuture in case of validation doesn't go through.